app/api/gsheets.py

```diff
@@ -0,0 +1,39 @@
+import click
+import re
+import gspread
+import datetime
+
+
+def csv_url_for_sheets_url(url):
+    """extract the parameters from a google docs url and formulate a CSV export url"""
+    m = re.search('.*\/d\/(.*)\/edit.*#gid=(.*)', url)
+    if m:
+        key = m.group(1)
+        gid = m.group(2)
+        return f"https://docs.google.com/spreadsheets/d/{key}/export?format=csv&gid={gid}"
+    else:
+        click.echo('Invalid Google Sheets URL')
+        raise click.Abort()
+
+
+def save_to_sheet(target_sheet_url, df):
+    """save df to the google doc at target_sheet_url. The data will always be saved to the first
```

Comment:
this is awesome! Question though - would it be easy to let the user specify both the read sheet and the write sheet directly? There might be cases where the tab we need to write to is different, or in a totally different sheet (looking at you FL)

Reply:
I can do that. Would it always be the first tab of the write sheet or should both the sheet and the tab be specified?

Reply:
Let's say both the sheet and the tab. (Thinking of the "Final" column in https://docs.google.com/spreadsheets/d/1Aph2HoNP2uoz7D0G5R5FioBBTgC4lEYCuTKdo4H0OlA/edit#gid=1101937167 as likely where the target full URLs would come from)

Reply:
That should be happening now. `--write-to-sheet` now takes the destination URL (including tab), as in `flask aggregate_outbreaks --write-to-sheet "https://docs.google.com/spreadsheets/d/1r7DU4FN_spe71nMa2lsIXAXgGEBw8mwyryiDlby3Q6w/edit#gid=1101937167" "https://docs.google.com/spreadsheets/d/1r7DU4FN_spe71nMa2lsIXAXgGEBw8mwyryiDlby3Q6w/edit#gid=273523772"` will read from the second URL and write the processed data to the first one. 

It's still making the backup copy of the tab before updating the data, which makes me feel better, but we might need to (optionally?) disable that to deal with the large sheets. It's a bit of a chore to delete the backups too.

I suspect there's a future revision to the CLI in which your sheet-of-sheets serves as the input that controls everything, but one thing at a time.